packages/sdk/src/utils/cache-singleton.ts

```diff
@@ -0,0 +1,32 @@
+export class CacheSingleton {
+  private static instance: CacheSingleton
+  private cache: Map<string, any>
```

Comment:
nah, don't use `any`

Reply:
you can use `unknown`

Reply:
or to have guaranteed on type-safety just make the cache singleton to hold the type, which has to be the same on setter and getter

Reply:
unknown causes issues with this:

export class CacheUtils {
  private static cache = CacheSingleton.getInstance()

  public static async getOrSetCache<T>(key: string, fetchData: () => Promise<T>): Promise<T> {
    let data = this.cache.get(key)
    if (!data) {
      data = await fetchData()
      this.cache.set(key, data)
    }
    return data
  }
}

how can I make this general purpose without any?